Tasks/PowerShellOnTargetMachines2/JobHelper.ps1

```diff
@@ -0,0 +1,33 @@
+function Run-RemoteScriptJobs {
+    [CmdletBinding()]
+    Param(
+        [System.Management.Automation.Runspaces.PSSession[]] $sessions,
+        [scriptblock] $script,
+        [System.Array] $scriptArguments,
+        [switch] $ignoreLASTEXITCODE
+    )
+    Trace-VstsEnteringInvocation -InvocationInfo $MyInvocation -Parameter "sessions","scriptArguments"
+    try {
+        $jobs = Invoke-Command -Session $sessions -AsJob -ScriptBlock $script -ArgumentList $scriptArguments
+        $jobResults = @();
+    
+        while($jobs | Where-Object { $_.State -eq "Running"}) {
+            ForEach($job in $jobs) {
+                Receive-Job -Job $job | ForEach-Object { if($_.VstsTask -eq $true) { $jobResults += $_ } else { Write-Host $($_ | Out-String) } }
+            }
+            Start-Sleep -Seconds 30
+        }
+    
+        ForEach($job in $jobs) {
```

Comment:
Is the output interleaved when executed on multiple jobs?

Reply:
yes

Reply:
an improvement we can make here is to 'tee' the output to a file as well. and at the end, upload those output files for each machine, so they may be downloaded as part of the release logs. @RoopeshNair  to take a call on this.